crates/compression/src/services/compress.rs

```diff
@@ -0,0 +1,87 @@
+use fuel_core_types::{
+    blockchain::block::Block,
+    fuel_compression::CompressibleBy,
+    fuel_tx::Transaction,
+};
+
+use crate::{
+    context::{
+        compress::CompressCtx,
+        prepare::PrepareCtx,
+    },
+    eviction_policy::CacheEvictor,
+    ports::{
+        TemporalRegistry,
+        UtxoIdToPointer,
+    },
+    tables::{
+        PerRegistryKeyspace,
+        RegistrationsPerTable,
+    },
+    CompressedBlockPayload,
+    Header,
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Only the next sequential block can be compressed")]
+    NotLatest,
+    #[error("Unknown compression error")]
+    Other(#[from] anyhow::Error),
+}
+
+pub trait CompressDb: TemporalRegistry + UtxoIdToPointer {}
+impl<T> CompressDb for T where T: TemporalRegistry + UtxoIdToPointer {}
+
+pub async fn compress<D: CompressDb>(db: D, block: &Block) -> Result<Vec<u8>, Error> {
+    // if *block.header().height() != db.next_block_height()? {
+    //     return Err(Error::NotLatest);
+    // }
+
+    let target = block.transactions().to_vec();
+
+    let mut prepare_ctx = PrepareCtx {
+        db,
+        accessed_keys: PerRegistryKeyspace::default(),
+    };
+    let _ =
+        <Vec<Transaction> as CompressibleBy<_>>::compress_with(&target, &mut prepare_ctx)
+            .await?;
+
+    let mut ctx = CompressCtx {
+        db: prepare_ctx.db,
+        cache_evictor: CacheEvictor {
+            keep_keys: prepare_ctx.accessed_keys,
+        },
+        changes: Default::default(),
+    };
```

Comment:
```suggestion
    let mut ctx: CompressCtx = prepare_ctx.into();
```

where 
```rust
impl<D: CompressDb> Into<CompressCtx<D>> for PrepareCtx<D> {
  fn into(self) -> CompressCtx<D> {
    CompressCtx {
      db: self.db,
      cache_evictor: CacheEvictor {
        keep_keys: self.accessed_keys,
      },
      changes: Default::default()
    }
  }
}
```

Reply:
I don't feel like hiding the behavior like this makes much sense, especially when it's only used in a single location.